tests/programs/test_api_update_program.py

```diff
@@ -0,0 +1,218 @@
+            "target_candidate_gender": "Other",
+            "target_candidate_age": "Not Applicable",
+            "target_candidate_ethnicity": "Not Applicable",
+            "target_candidate_sexual_orientation": "Not Applicable",
+            "target_candidate_religion": "Not Applicable",
+            "target_candidate_physical_ability": "Not Applicable",
+            "target_candidate_mental_ability": "Not Applicable",
+            "target_candidate_socio_economic": "Not Applicable",
+            "target_candidate_highest_education": "Not Applicable",
+            "target_candidate_years_of_experience": "Not Applicable",
+            "target_candidate_other": "Non-Binary",
+        }
+        program.contact_type = "REMOTE"
+        program.zone = "GLOBAL"
+        program.status = "DRAFT"
+        program.creation_date = 1601424000
+        program.save_to_db()
+
+        self.program_data1 = ProgramModel.find_by_name("Program A")
+
+
+    def test_api_dao_update_program_successfully(self):
+        # prepare existing program with 
+        # start date 2020-10-15 12:00 GMT+0
+        # and end date 2020-11-25 12:00 GMT+0
+
+        test_user_extension = UserExtensionModel(
+            user_id=self.test_user1_data.id,
+            timezone="AUSTRALIA_MELBOURNE"
+        )
+        test_user_extension.is_organization_rep = True
+        test_user_extension.save_to_db()
+
+        response = self.client.put(
+            f"/organizations/{self.test_organization_data.id}/programs/{self.program_data1.id}",
+            headers={"Authorization": AUTH_COOKIE["Authorization"].value},
+            data=json.dumps(
+                dict(self.update_payload_program)
+                ),
+            follow_redirects=True,
+            content_type="application/json",
+            )
+        self.assertEqual(HTTPStatus.OK, response.status_code)
+        self.assertEqual(messages.PROGRAM_SUCCESSFULLY_UPDATED, response.json)
+
+    def test_api_dao_update_program_with_invalid_organization(self):
+        self.organization_id = 3
+
+        test_user_extension = UserExtensionModel(
+            user_id=self.test_user1_data.id,
+            timezone="AUSTRALIA_MELBOURNE"
+        )
+        test_user_extension.is_organization_rep = True
+        test_user_extension.save_to_db()
+
+        response = self.client.put(
+            f"/organizations/{self.organization_id}/programs/{self.program_data1.id}",
+            headers={"Authorization": AUTH_COOKIE["Authorization"].value},
+            data=json.dumps(
+                dict(self.update_payload_program)
+                ),
+            follow_redirects=True,
+            content_type="application/json",
+            )
+        print(response)
+        self.assertEqual(HTTPStatus.NOT_FOUND, response.status_code)
+        self.assertEqual(messages.ORGANIZATION_DOES_NOT_EXIST, response.json)
+
+    def test_api_dao_update_program_not_representative(self):
+        response = self.client.put(
```

Comment:
The logic in this test a bit off. As part of the Program logic, the user need to declare they represent organization first before they can create an organization then create a program. This means to create an organization and program they must alrady have user_extension instance in the database (You can see in your test setup() that the organization is using user1 id as the representative). So, this test block (user is not the rep of organization so is not allowed to update program), although resulted in pass, this is not something that will ever happen in the actual application.

Other way to test this point is to create another organization and another program in this block of test and use that organization and program to test if user1 can send this update program request. Also, with this logic, you can move the user1 user_extension initiation (line 164 and 186) to the setup() block.